### PR TITLE
chore(hapi): update hapi ecosystem to ts 3.x

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/hapijs/hoek/blob/master/API.md https://github.com/hapijs/wreck
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

This touched a few packages across the @hapi namespace to improve type safety.
Core changes are in hapi and hoek.

### CODE REVIEW

**Comment by SimonSchick** [comment]
Preemtively @lenovouser @AJamesPhillips

**Comment by AJamesPhillips** [comment]
I gave it a quick scan and nothing obvious jumped out but there's quite a bit in there that wasn't familiar to me anymore anyway @SimonSchick  :)

**Review by AJamesPhillips — APPROVED** [review]

File: types/hapi__hapi/test/server/server-cache.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // https://github.com/hapijs/hapi/blob/master/API.md#-servercacheoptions
 import { Server, ServerOptionsCache } from "@hapi/hapi";
-import * as catbox from "@hapi/catbox";
+import { PolicyOptions } from '@hapi/catbox';
```
Comment (saboya) [inline types/hapi__hapi/test/server/server-cache.ts]:
```suggestion
import { PolicyOptions } from "@hapi/catbox";
```

Either this should be double-quoted or the other import should be single-quoted I guess?

**Review by saboya — APPROVED** [review]

File: types/hapi__hapi/test/server/server-cache.ts (hunk 1 above)
Comment (SimonSchick) [reply types/hapi__hapi/test/server/server-cache.ts]: It's something my IDE did, I will see if I can consolidate if the header is the only inconsistency but it's out of scope of this PR.

**Review by todd — APPROVED** [review]
Good to go from my perspective, but my area of responsibility is only getting a TS version bump so ¯\\_(ツ)_/¯

**Review by lenovouser — APPROVED** [review]
Looks good to me so far

**Comment by orta** [comment]
👋  - happy to merge this, but switching a lot of any to unknowns can break builds, it might be worth making sure anything which got a `2.x` -> `3.1` bump gets a major server bump in the types?

**Review by mrcrane — CHANGES_REQUESTED** [review]
Marking as changes requested until @orta's suggestion is addressed.

**Comment by SimonSchick** [comment]
@orta @mrcrane This PR might be mood anyways since the hapi contributors are going to ship their own typings soon (they have already started with some libraries) with MUCH MUCH more breaking changes.

**Pull request closed without merging**